### PR TITLE
Feature/stability

### PR DESCRIPTION
**Pull request by pogobanane**
- add vm-memory atomicity (didn't really help)
- handle irq ack timeouts (helped)

i saw irq drop rates of 1-4% in access-intensive scenarios
`re-sending lost interrupt after 2.2ms. Total lost 1% (2327/219493)`

### CODE REVIEW

**Comment by pogobanane** [comment]
there are some `.expect("")`s to fix

File: src/device/threads.rs [hunk 1]
```diff
@@ -97,40 +109,57 @@ fn blkdev_monitor_thread(
...
+                    //     blkdev.selected_queue().unwrap().is_valid() // crashes before activate()
+                    // );
+                    // debug!(
+                    //     "dev queue {}: avail idx {}, next avail idx {}",
+                    //     blkdev.queue_select(),
+                    //     blkdev.selected_queue().unwrap().avail_idx(Ordering::Relaxed).unwrap(),
+                    //     blkdev.selected_queue().unwrap().next_avail(),
+                    //     );
+                    debug!(
+                        "dev queue {}: irq status b{:b}",
```
Comment (Mic92) [inline src/device/threads.rs:149]: Was the code above removed on purpose?

**Comment by Mic92** [comment]
Good job!

File: src/device/threads.rs (hunk 1 above)
Comment (pogobanane) [reply src/device/threads.rs:149]: Yes. I have not found any use for this information yet. It just clogs the output. But i also don't want to delete it.

**Pull request merged**